### PR TITLE
test: Upgrade execution tests to wat2wasm (2)

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: test/unittests/execute_test.cpp [hunk 1]
```diff
@@ -93,68 +93,58 @@ TEST(execute, local_tee)
...
-
-    const auto [trap, ret] = execute(instance, 0, {});
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], 42);
+    /* wat2wasm
+    (global i32 (i32.const 42))
+    (func (result i32)
+      get_global 0
```
Comment (axic) [inline test/unittests/execute_test.cpp:100]: Curious, why are we not using named fields in any of the tests?

File: test/unittests/execute_test.cpp [hunk 2]
```diff
@@ -93,68 +93,58 @@ TEST(execute, local_tee)
...
+    (func (result i64)
+      get_global 0
+    )
+    */
+    const auto wasm = from_hex(
+        "0061736d010000000105016000017e020d01036d6f6404676c6f62037e00030201000a0601040023000b");
+    const auto module = parse(wasm);
 
-    uint64_t global_value = 42;
+    uint64_t global_value = 0;
```
Comment (axic) [inline test/unittests/execute_test.cpp:141]: Why not keep this test case?

You could also add `EXPECT_RESULT(execute(), 0);` if you change this.

File: test/unittests/execute_test.cpp [hunk 3]
```diff
@@ -843,143 +843,115 @@ TEST(execute, start_section)
...
 {
-    Module module;
-    module.typesec.emplace_back(FuncType{{ValType::i32, ValType::i32}, {ValType::i32}});
-    module.importsec.emplace_back(Import{"mod", "foo", ExternalKind::Function, {0}});
+    /* wat2wasm
+    (import "mod" "foo" (func (param i32 i32) (result i32)))
+    */
+    const auto wasm = from_hex("0061736d0100000001070160027f7f017f020b01036d6f6403666f6f0000");
+    const auto module = parse(wasm);
+    ASSERT_EQ(module.typesec.size(), 1);
```
Comment (axic) [inline test/unittests/execute_test.cpp:625]: Why is this needed?

File: test/unittests/execute_test.cpp [hunk 4]
```diff
@@ -843,143 +843,115 @@ TEST(execute, start_section)
...
-    module.typesec.emplace_back(FuncType{{ValType::i64}, {ValType::i64}});
-    module.importsec.emplace_back(Import{"mod", "foo1", ExternalKind::Function, {0}});
-    module.importsec.emplace_back(Import{"mod", "foo2", ExternalKind::Function, {0}});
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.codesec.emplace_back(Code{0, {Instr::i32_const, Instr::end}, {42, 0, 42, 0}});
+    /* wat2wasm
+    (type (func (param i32 i32) (result i32)))
+    (type (func (param i64) (result i64)))
+    (import "mod" "foo1" (func (type 0)))
+    (import "mod" "foo2" (func (type 1)))
```
Comment (axic) [inline test/unittests/execute_test.cpp:705]: It seems the original test case was buggy and always used type 0.

File: test/unittests/execute_test.cpp [hunk 5]
```diff
@@ -240,36 +240,37 @@ TEST(execute, global_set_imported)
...
 {
-    Module module;
-    Import imp{"mod", "m", ExternalKind::Memory, {}};
-    imp.desc.memory = Memory{{1, 1}};
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.importsec.emplace_back(imp);
-    module.codesec.emplace_back(
-        Code{0, {Instr::local_get, Instr::i32_load, Instr::end}, {0, 0, 0, 0, 0, 0, 0, 0}});
+    /* wat2wasm
+    (import "mod" "m" (memory 1))
```
Comment (axic) [inline test/unittests/execute_test.cpp:244]: Maybe make these explicit `memory 1 1`?

File: test/unittests/execute_numeric_test.cpp [hunk 6]
```diff
@@ -30,29 +30,23 @@ execution_result execute_binary_operation(Instr instr, uint64_t lhs, uint64_t rh
...
+    /* wat2wasm
+    (func (result i64) (i64.const 0x0100000000420042))
+    */
+    const auto wasm =
+        from_hex("0061736d010000000105016000017e030201000a0e010c0042c280888280808080010b");
 
-    ASSERT_EQ(trap, false);
-    ASSERT_EQ(ret.size(), 1);
-    EXPECT_EQ(ret[0], 0x0100000000420042ULL);
+    EXPECT_RESULT(execute(parse(wasm), 0, {}), 0x0100000000420042);
```
Comment (axic) [inline test/unittests/execute_numeric_test.cpp:49]: My compiler was complaining (when the test was written) and needed the `ULL`  annotation.

Comment (axic) [reply test/unittests/execute_numeric_test.cpp:49]: Tried now, this PR works without it.

File: test/unittests/execute_test.cpp (hunk 5 above)
Comment (chfast) [reply test/unittests/execute_test.cpp]: Changed.

File: test/unittests/execute_test.cpp (hunk 2 above)
Comment (chfast) [reply test/unittests/execute_test.cpp]: Restored original case and added one more with resetting global value to 0.

File: test/unittests/execute_test.cpp [hunk 7]
```diff
@@ -762,226 +776,183 @@ TEST(execute, i64_store32)
...
+    EXPECT_RESULT(execute(instance, 0, {20, 22}), 42);
+    EXPECT_RESULT(execute(instance, 1, {20}), 400);
 
     // check correct number of arguments is passed to host
-    auto count_args = [](Instance&, std::vector<uint64_t> args) -> execution_result {
+    constexpr auto count_args = [](Instance&, std::vector<uint64_t> args) -> execution_result {
         return {false, {args.size()}};
     };
 
     auto instance_couner =
```
Comment (gumb0) [inline test/unittests/execute_test.cpp:920]: Could you fix the typo
```suggestion
    auto instance_counter =
```

File: test/unittests/execute_test.cpp [hunk 8]
```diff
@@ -762,226 +776,183 @@ TEST(execute, i64_store32)
...
-    EXPECT_EQ(ret2[0], 400);
-
-    const auto [trap3, ret3] = execute(instance, 2, {20});
+        instantiate(module, {{host_foo1, module.typesec[0]}, {host_foo2, module.typesec[1]}});
 
-    ASSERT_FALSE(trap3);
-    ASSERT_EQ(ret3.size(), 1);
-    EXPECT_EQ(ret3[0], 0x2a002a);
+    EXPECT_RESULT(execute(instance, 0, {20, 22}), 42);
+    EXPECT_RESULT(execute(instance, 1, {0x30000000}), 0x900000000000000);
```
Comment (gumb0) [inline test/unittests/execute_test.cpp:954]: will this compile?
```suggestion
    EXPECT_RESULT(execute(instance, 1, {0x3000'0000}), 0x900'0000'0000'0000);
```
